### PR TITLE
Add HUD + Player controls + "All Finished" state

### PR DESCRIPTION
**Pull request by peteygao**
Added the HUD at the top of the Player. It shows the current Guide Name + Total Guides and the elapsed time in the current Guide. The timer stops once the "Finished" step is reached for the current Guide.

The player controls are the left/right arrow at the bottom which will navigate between the available Guides.

And additional "AllFinished" global Guide is created for when all of the Guides are done. The global All Finished instruction is displayed instead of the Finished instruction for the final Guide.

### CODE REVIEW

File: Guides/Global/AllFinished.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+
+export default class AllFinished extends Component {
```
Comment (artemkhere) [inline Guides/Global/AllFinished.js]: Maybe consider renaming this to be `FinishedGuide` or `GuideFinished`? I think `AllFinished` is a little ambiguous, especially if we don't end up using `All Finished` as a copywriting for that slide.

File: SessionPlayer/HUD.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    componentWillUnmount() {
+        clearInterval(this.timerIntervalID);
+    }
+
+    componentDidUpdate() {
+        const { guideState } = this.props;
+
+        if (
+            this.timerIntervalID
+            && (guideState === 'finished' || guideState === 'all_finished')
```
Comment (artemkhere) [inline SessionPlayer/HUD.js]: I think `all_finished` is an ambiguous state. I believe `guide_finished` makes a little more sense (or something similar).

File: SessionPlayer/HUD.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        ) {
+            clearInterval(this.timerIntervalID);
+        }
+    }
+
+    render() {
+        const { currentGuideNumber, totalGuides, guideName } = this.props;
+        const { timer } = this.state;
+
+        const minutes = Math.floor(timer / 60);
```
Comment (artemkhere) [inline SessionPlayer/HUD.js:49]: Can we move this out of the `render` function , please?

File: SessionPlayer/HUD.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        const seconds = timer % 60;
+
+        return (
+            <View style={styles.container}>
+                <Text style={styles.guideName}>
+                    {guideName}
+                </Text>
+                <View style={styles.smallTextContainer}>
+                    <Text style={styles.text}>
```
Comment (artemkhere) [inline SessionPlayer/HUD.js]: I think rendering both of `current guide` and `timer` should live in their own functions. So `render` could just put them together like:
```
return (
            <View style={styles.container}>
                <Text style={styles.guideName}>
                    {guideName}
                </Text>
                {this.renderGuideNumber()}
                {this.renderGuideTimer()}
            </View>
```
or something like that.

File: SessionPlayer/SessionPlayer.js [hunk 5]
```diff
@@ -59,7 +61,7 @@ export default class SessionPlayer extends Component {
             ];
 
         this.state = {
-            guideState: 'playing', // 'playing', 'paused', 'finished'
+            guideState: 'playing', // 'playing', 'paused', 'finished', 'all_finished'
```
Comment (artemkhere) [inline SessionPlayer/SessionPlayer.js]: Again, I don't like `all_finished`

File: SessionPlayer/SessionPlayer.js [hunk 6]
```diff
@@ -92,20 +95,70 @@ export default class SessionPlayer extends Component {
...
             this.setState({
+                guideState: 'playing',
                 currentInstructionIndex: 0,
                 currentInstruction: nextGuide.instructions[0],
                 currentGuideIndex: nextGuideIndex,
                 currentGuide: nextGuide,
             });
         } else {
+            this.setState({
+                guideState: 'all_finished',
```
Comment (artemkhere) [inline SessionPlayer/SessionPlayer.js]: No bueno

File: SessionPlayer/HUD.js (hunk 2 above)
Comment (artemkhere) [reply SessionPlayer/HUD.js]: No, I think it needs to be `session_finished`.

File: Guides/Global/AllFinished.js (hunk 1 above)
Comment (artemkhere) [reply Guides/Global/AllFinished.js]: Actually, I think it needs to be `SessionFinished`

File: SessionPlayer/SessionPlayer.js [hunk 7]
```diff
@@ -121,76 +174,118 @@ export default class SessionPlayer extends Component {
...
+                                + currentInstructionIndex.toString()
+                                + this.state.guideState
+                            }
                             onInstructionEnd={this.onInstructionEndHandler}
                             duration={currentInstruction.duration}
                             instructionsText={currentInstruction.instructionsText}
                         />
                     </View>
                     <View style={styles.actionButtonContainer}>
+                        <Icon
```
Comment (artemkhere) [inline SessionPlayer/SessionPlayer.js]: This long `render` makes a lot of sense. It is putting right components and correct props together, however the `<Icon>` construction involves sizing, icon name and etc. I don't think render needs to know about all of it. Something like `renderDecrementGuideButton` should be more appropriate IMO.

**Review by artemkhere — COMMENTED** [review]
Overall, it is looking great!

**Comment by peteygao** [comment]
@artemkhere Addressed all your comments :smile:!

**Pull request merged**